h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoReader.java

```diff
@@ -37,29 +34,26 @@ protected XGBoostMojoModel makeModel(String[] columns, String[][] domains, Strin
     } catch (IOException e) {
       throw new IllegalStateException("MOJO is corrupted: cannot read the serialized Booster", e);
     }
-    Boolean conf = getJavaScoringConfig();
-    if (useJavaScoring(conf)) {
+    Boolean javaScoringByDefault = readkv("use_java_scoring_by_default");
+    String booster = readkv("booster");
+    if (useJavaScoring(javaScoringByDefault, booster)) {
       return new XGBoostJavaMojoModel(boosterBytes, columns, domains, responseColumn);
     } else {
       return new XGBoostNativeMojoModel(boosterBytes, columns, domains, responseColumn);
     }
   }
 
-  private boolean useJavaScoring(Boolean conf) {
-    if (conf != null) {
-      // user set the property - respect the decision
-      return conf;
-    }
-    String booster = readkv("booster");
-    return "gbtree".equals(booster); // use java scoring for `gbtree` (well tested); `native` for dart & gblinear
-  }
-
-  public static Boolean getJavaScoringConfig() {
+  public static boolean useJavaScoring(Boolean mojoValue, String booster) {
```

Comment:
imho we can call it mojoPreferredJavaScoring (or something like that ), I wasn't sure what mojoValue meant when I saw this method

Reply:
fixed